python/graphstorm/sagemaker/sagemaker_train.py

```diff
@@ -308,7 +303,97 @@ def run_train(args, unknownargs):
         logging.error("Task failed")
         sys.exit(-1)
 
-    # We upload models only when the user explicitly set the model_artifact_s3
-    # argument. Otherwise we can rely on the SageMaker service to do the upload.
-    if gs_model_artifact_s3 and os.path.exists(save_model_path):
-        upload_model_artifacts(gs_model_artifact_s3, save_model_path, sagemaker_session)
+    # Copy model+embeddings from last epoch into local SageMaker output directory
+    # TODO: Support packing the best epoch from the run
+    copy_best_model_to_sagemaker_output(save_model_path, best_epoch=None)
+
+
+def copy_best_model_to_sagemaker_output(save_model_path, best_epoch=None):
+    """Copy the best or latest epoch model and config files to SageMaker's
+       standard model output directory.
+
+    Parameters
+    ----------
+    save_model_path: str
+        Path to the directory containing existing model checkpoints
+    best_epoch: str, optional
+        Name of the best epoch directory (e.g., 'epoch-5'). If None, the latest epoch will be used.
+    """
+    # TODO: Do we want to be moving files instead for improved perf?
+    if not os.path.exists(save_model_path):
+        logging.warning("Model path %s does not exist, nothing to copy",
+                        save_model_path)
+        return
+
+    # If best_epoch is provided, try to use it directly
+    if best_epoch is not None:
+        best_epoch_dir = best_epoch
+        best_epoch_path = os.path.join(save_model_path, best_epoch_dir)
+        if not os.path.exists(best_epoch_path):
+            logging.warning(
+                "Best epoch directory %s does not exist, falling back to latest epoch",
+                best_epoch_path)
+            best_epoch = None
+
+    # If best_epoch was not provided or not found, find the latest epoch
+    if best_epoch is None:
```

Comment:
It is better to have some unitest to guard the implementation.

Reply:
I'll a unit test with Q, this file deals with local data (no S3) so should be easy to mock